### DISCUSSION TITLE
EJEMPLO DE MVC

### DISCUSSION THREAD

**Discussion by habacucsoto** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#8] [Show and tell]
![IMG_20220216_195641](https://user-images.githubusercontent.com/86739811/154390062-bd3f5e60-0631-46d4-b02c-c4381a180177.jpg)
Donde back es modelo, front es vista y api controlador

**Comment by RicardoRobledo** [comment]
Imagina que tienes una base de datos para punto de venta de una farmacia, la parte de modelo o "M" son las entidades de tu base de datos que posteriormente tienes que definir en tu programa como clases, por ejemplo, una entidad de tu base de datos sería un artículo como un frasco de vitaminas en el que "Artículo" es una tabla en tu base de datos, pero esas entidades solo existen en tu base de datos que puede encontrarse en línea o en algún servidor, ¿Cómo puedes hacer que tu programa use las entidades para poder pedir los datos que se requieren para un nuevo registro?, bueno en tu programa debes definir las entidades que existen también en tu base de datos, no hay de otra porque debes de interactuar con clases y objetos como es tener una clase en tu programa de la tabla "Artículo" para que se pueda tener una plantilla y crear objetos diferentes o artículos diferentes que le lleguen para su inventario donde el usuario con ayuda de la vista o "V" siendo las interfaces visuales pueda ingresar los datos del producto como el nombre del producto, su costo, en pocas palabras los mismos datos que se requieren para un registro en tu base de datos, una vez ingresados los datos imagina que dan click a un botón de alta de artículo, en la vista se supone que te pidieron datos para poder registrar un nuevo artículo, con esos datos se crea un objeto de tipo "Artículo" que a su vez se pasa al controlador o "C", este controlador sirve de intermediario para la vista y el modelo, en pocas palabras una vez mandas a llamar al controlador ese se encarga del proceso de alta, es decir ir a la base de datos y registrar tu nuevo artículo. Espero que te haya servido.
